kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -223,6 +260,25 @@ protected final Optional<List<FileStatus>> listDeltaAndCheckpointFiles(
                         .orElse(true);
 
                     if (!versionWithinRange) {
+                        // If we haven't taken any files yet and the first file we see is greater
+                        // than the versionToLoad then the versionToLoad is not reconstructable
+                        // from the existing logs
+                        if (output.isEmpty()) {
+                            // TODO this should be a public exception
+                            // We can include the log retention and checkpoint retention confs
+                            // to provide more information
+                            // (see logFileNotFoundException in DeltaErrors)
+                            throw new RuntimeException(
+                                String.format(
+                                    "Unable to reconstruct state at version %s as the " +
+                                        "transaction log has been truncated due to manual " +
+                                        "deletion or the log retention policy and checkpoint " +
+                                        "retention policy.",
```

Comment:
i thought we'd be using `DeltaErrors.nonReconstructableState(...)` or something like that? to remove these verbose strings from our code and refactor them to a separtate file?